ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -119,13 +121,17 @@ public void sendRequest(
       final int requestId,
       ReadableArray headers,
       ReadableMap data,
-      final boolean useIncrementalUpdates) {
+      final boolean useIncrementalUpdates,
+      int timeout) {
     Request.Builder requestBuilder = new Request.Builder().url(url);
 
     if (requestId != 0) {
       requestBuilder.tag(requestId);
     }
 
+    // requestBuilder.setTimeoutMillis(timeout);
+    mClient.setConnectTimeout(timeout, TimeUnit.MILLISECONDS);
```

Comment:
Looks like 0 means no timeout: https://square.github.io/okhttp/2.x/okhttp/com/squareup/okhttp/OkHttpClient.html#setConnectTimeout-long-java.util.concurrent.TimeUnit-

Please remove the commented out line.